chain/transaction_test.go

```diff
@@ -115,17 +115,20 @@ func TestMarshalUnmarshal(t *testing.T) {
 	err = actionRegistry.Register(&action2{}, unmarshalAction2)
 	require.NoError(err)
 
+	txBeforeSign := tx
+	require.Nil(tx.Auth)
 	signedTx, err := tx.Sign(factory, actionRegistry, authRegistry)
 	require.NoError(err)
-
+	require.Equal(txBeforeSign, tx)
```

Comment:
Is it possible for `require.Equal(txBeforeSign, tx)` to fail even if it were still modifying the tx? I think this is checking the same pointer

Reply:
given that we don't have a deep clone method for the Transaction, I'll just create the "expected" copy of the transaction definition and compare against that.